DuckDuckGo/MainViewController.swift

```diff
@@ -998,6 +982,7 @@ class MainViewController: UIViewController {
         currentTab?.progressWorker.progressBar = nil
         currentTab?.chromeDelegate = nil
             
+        tab.delegate = self
```

Comment:
I see what you mean by delaying the moment delegate is attached. The problem here is that before delegate is set, the view is apparently already loaded. The most likely place that could fail are the ones that use delegate methods which return a value like `tabDidRequestSearchBarRect` or `tabCheckIfItsBeingCurrentlyPresented` which could glitch the TabUI itself during its setup depending on the code flows (there could be other glitches too, but these are most likely). 

We could investigate code paths to "fix" this now, but overall this is not a foolproof solution. I'm afraid that with TabManager being a Factory, a binding must occur as it was.

But looking at the code, I think it could remain as it was - just maybe don't pass self it through TabManager init as it's awkward for weak vars but explicitly invoke a manager.tabDelegate = self in MVC init.